### PR TITLE
FIR IDE: add quickfix for API mode related diagnostics

### PR DESCRIPTION
**Pull request by tgeng**

### CODE REVIEW

File: plugins/kotlin/frontend-independent/src/org/jetbrains/kotlin/idea/util/psiUtils.kt [hunk 1]
```diff
@@ -135,3 +137,56 @@ tailrec fun KtCallableDeclaration.isExplicitTypeReferenceNeededForTypeInference(
...
+        val parameter = valueParameters.singleOrNull() ?: return false
+        val binaryExpression = statement as? KtBinaryExpression ?: return false
+        return binaryExpression.operationToken == KtTokens.EQ
+                && binaryExpression.left?.isBackingFieldReferenceTo(property) == true
+                && binaryExpression.right?.mainReference?.resolve() == parameter
+    }
+    return false
+}
+
+fun removeRedundantSetter(setter: KtPropertyAccessor) {
```
Comment (jsjeon) [inline plugins/kotlin/frontend-independent/src/org/jetbrains/kotlin/idea/util/psiUtils.kt:155]: Nit: what about moving `removeRedundantGetter` too?

File: plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/ChangeVisibilityFixFactories.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            createFixForNoExplicitVisibilityInApiMode(diagnostic.psi)
+        }
+    val noExplicitVisibilityInApiModeWarning =
+        diagnosticFixFactory(KtFirDiagnostic.NoExplicitVisibilityInApiModeWarning::class, makePublicExplicitApplicator) { diagnostic ->
+            createFixForNoExplicitVisibilityInApiMode(diagnostic.psi)
+        }
+
+    private fun createFixForNoExplicitVisibilityInApiMode(declaration: KtDeclaration): List<HLApplicatorTargetWithInput<KtDeclaration, Input>> {
+        val name = when (declaration) {
+            is KtConstructor<*> -> "<init>"
```
Comment (jsjeon) [inline plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/ChangeVisibilityFixFactories.kt:62]: We don't have a centralized place for this special name?!

File: plugins/kotlin/frontend-independent/src/org/jetbrains/kotlin/idea/util/psiUtils.kt (hunk 1 above)
Comment (tgeng) [reply plugins/kotlin/frontend-independent/src/org/jetbrains/kotlin/idea/util/psiUtils.kt:155]: Ah, yeah, why not?

File: plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/ChangeVisibilityFixFactories.kt (hunk 2 above)
Comment (tgeng) [reply plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/ChangeVisibilityFixFactories.kt:62]: Good point! Actually we do in `SpecialNames`. Let me use that instead.

**Review by jsjeon — APPROVED** [review]

**Comment by darthorimar** [comment]
Merged, thank you :)

**Pull request closed without merging**